server/pull_request.go

```diff
@@ -535,3 +527,15 @@ func (s *Server) getPRFromComment(ctx context.Context, comment IssueComment) (*m
 	}
 	return pr, nil
 }
+
+func (s *Server) setBlockStatusForPR(ctx context.Context, pr *model.PullRequest) {
```

Comment:
shall we propagate error here as well?

Reply:
I thought about it but if we propagate the error then we're going to have multiple `mlog.Error` with the same message because that error is not returning a 4xx error by design so if the outcome of that error is showing an error trace then should be better be encapsulated inside that method